apps/verticalhome/js/app.js

```diff
@@ -150,6 +179,177 @@
+                pinnedItems[i].entry.icon.baseURI
+                :pinnedItems[i].entry.icon;
+
+              obj.index = pinnedItems[i].index;
+
+              app.itemStore.savePinnedAppItem(obj);
+
+          }
+        }
+
+       this.showNotif('Changes saved');
+
+    },
+
+    pinToMainScreen: function(){
+      var moreAppsSelected = '#more-apps-screen .selected';
+      var selMoreApp = document.querySelector(moreAppsSelected);
+      var selMoreAppIcon = selMoreApp.getAttribute('data-test-icon');
+      var selMoreAppDataId = selMoreApp.getAttribute('data-identifier');
+      var selMoreAppTitle = selMoreApp.querySelector('span.title').textContent;
+      var maxDataIndex = app.pinnedAppsManager.items.length;
+
+      app.pinnedAppsManager.addEntry( maxDataIndex,
+                                      selMoreAppDataId,
+                                      selMoreAppIcon,
+                                      selMoreAppTitle);
+
+      Array.prototype.slice.call(
+        document.querySelector('#more-apps-screen #icons').children
+      );
+
+      app.itemStore.applicationSource.
+        removeIconFromGrid( selMoreAppDataId );
+
+      this.renderGrid();
+
+      if (selMoreApp.parentNode) {
+        selMoreApp.parentNode.removeChild(selMoreApp);
+      }
+
+      this.hideMoreApps();
+
+      this.showNotif('App added to Home Screen');
+
+      var lastTimeSelected = app.pinnedAppsNavigation.currentIndex();
+      var insrertedIndex = app.pinnedAppsManager.items.length;
+      var scrollDiff = insrertedIndex - maxDataIndex;
+      var selectedPinnedApp = app.pinnedAppsNavigation
+                              .elements[lastTimeSelected];
+
+      selectedPinnedApp.classList.remove('selected');
+
+      for(var z = 0; z < scrollDiff; z++){
+        var itm = app.pinnedAppsNavigation.elements[0];
+        var cln_tmp = itm.cloneNode(true);
+        app.pinnedAppsNavigation.elemList.appendChild(cln_tmp);
+        itm.parentNode.removeChild(itm);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+
+      app.pinnedAppsNavigation.scrollNum();
+
+      app.pinnedAppsNavigation.refresh();
+
+    },
+
+    unpinFromMainScreen: function(){
+
+      var selPinApp = document.querySelector('#pinned-apps-list .selected');
+      var selPinAppDataId = selPinApp.getAttribute('data-index');
+
+      app.pinnedAppsNavigation.refresh();
+
+      this.deletedApps.push(
+        app.pinnedAppsManager.items[selPinAppDataId]
+        );
+
+      app.pinnedAppsManager.items[selPinAppDataId].
+        element.style.display = 'none';
+
+      if (selPinApp.parentNode) {selPinApp.parentNode.removeChild(selPinApp);}
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+      app.pinnedAppsNavigation.elemList.classList.add('animation');
+      app.pinnedAppsNavigation.refresh();
+
+      this.showNotif('App removed from Homescreen and moved to "More apps"');
+
+    },
+
+    personalize: function(){
```

Comment:
maybe remove to something like ``startPersonalize`` to align with ``endPersonalize`` below?